dns/resolver.go

```diff
@@ -186,16 +186,16 @@ func queryDatagram(conn io.ReadWriter, q dnsmessage.Question) (*dnsmessage.Messa
 			err = nil
 		}
 		if err != nil {
-			return nil, &nestedError{ErrReceive, errors.Join(returnErr, fmt.Errorf("read message failed: %w", err))}
+			return nil, &nestedError{ErrReceive, wrapError(returnErr, fmt.Errorf("read message failed: %w", err))}
```

Comment:
The semantic here is broken.
These are not nested errors. They are sibling.
You are instead saying that the error is the last one, with unrelated sub-errors.

We need to define what semantic we want to represent in the UDP test.
Perhaps we only look at the first error? Call `Unwrap() []error` instead, extract the first one, then do the usual error handling.

If you need to look at the other errors, you need to figure out what that means. Still, that can live in the test code.